Analysis/PIDTools/src/LikelihoodPIDProcessor.cc

```diff
@@ -274,99 +309,63 @@ void LikelihoodPIDProcessor::processEvent( LCEvent * evt ) {
-    }
-    
-    //create PIDHandler
-    createParticleIDClass(parttype, part, pidh, algoID1, MVAoutput);
-
-//////////////////////////////////////////////////////////////////////////////////
-    //use just dEdx variables  (Algorithm 2)
-    _myPID->setBasicFlg(false);
-    _myPID->setdEdxFlg(true);
-    _myPID->setShowerShapesFlg(false);
-    parttype = _myPID->Classification(pp, trk, clu);
-    if(parttype<0) parttype=2;
-
-     //mu-pi Separation for very low momentum tracks (from 0.2 GeV until 2 GeV)
-    MVAoutput = -100.0;
-    if((parttype==1 || parttype==2) && (_UseMVA && pp.P()<2.0 && clu.size()!=0)){
-      int tmpparttype = _mupiPID->MuPiSeparation(pp, trk, clu);
-      if(_mupiPID->isValid()) parttype = tmpparttype;
-      MVAoutput = _mupiPID->getMVAOutput();   
-    }
-    
-    //create PIDHandler
-    createParticleIDClass(parttype, part, pidh, algoID2, MVAoutput);
-
-//////////////////////////////////////////////////////////////////////////////////
-    //use just Shower Profile variables  (Algorithm 3)
-    _myPID->setBasicFlg(false);
-    _myPID->setdEdxFlg(false);
-    _myPID->setShowerShapesFlg(true);
-    parttype = _myPID->Classification(pp, trk, clu);
-    if(parttype<0) parttype=2;
-    
-    //mu-pi Separation for very low momentum tracks (from 0.2 GeV until 2 GeV)
-    MVAoutput = -100.0;
-    if((parttype==1 || parttype==2) && (_UseMVA && pp.P()<2.0 && clu.size()!=0)){
-      int tmpparttype = _mupiPID->MuPiSeparation(pp, trk, clu);
-      if(_mupiPID->isValid()) parttype = tmpparttype;
-      MVAoutput = _mupiPID->getMVAOutput();   
-    }
-    
-    //create PIDHandler
-    createParticleIDClass(parttype, part, pidh, algoID3, MVAoutput);
-
-//////////////////////////////////////////////////////////////////////////////////
-    //calculate global likelihood (Algorithm 4)
-    _myPID->setBasicFlg(true);
-    _myPID->setdEdxFlg(true);
-    _myPID->setShowerShapesFlg(true);
-    parttype = _myPID->Classification(pp, trk, clu);
-    if(parttype<0) parttype=2;
-
-    //mu-pi Separation for very low momentum tracks (from 0.2 GeV until 2 GeV)
-    MVAoutput = -100.0;
-    if((parttype==1 || parttype==2) && (_UseMVA && pp.P()<2.0 && clu.size()!=0)){
-      int tmpparttype = _mupiPID->MuPiSeparation(pp, trk, clu);
-      if(_mupiPID->isValid()) parttype = tmpparttype;
-      MVAoutput = _mupiPID->getMVAOutput();   
-    }
-
-    //create PIDHandler
-    createParticleIDClass(parttype, part, pidh, algoID4, MVAoutput);
-
-//////////////////////////////////////////////////////////////////////////////////
-
-    //Low momentum Muon identification  (Algorithm 5)
-    // (from 0.2 GeV until 2 GeV)   
-    if(_UseMVA){
-      MVAoutput = -100.0;
-      parttype = -1;   
-      if(pp.P()<2.5){
-	parttype=_mupiPID->MuPiSeparation(pp, trk, clu);
-	MVAoutput = _mupiPID->getMVAOutput();   
+    for(int ialgo=0; ialgo<_methodstorun.size(); ialgo++) {
+      if(_methodstorun.at(ialgo).compare("LowMomID")==0) {
```

Comment:
Just to be sure here that `"LowMomID"` here and `"LowMomMuID"` below is not a typo.

Reply:
It is a typo!
Thanks!

it should be: LowMomMuID

Reply:
And it should be 
if(_methodstorun.at(ialgo) == "LowMomMuID")

Reply:
changes done and committed (see)  83b4df0